pkg/cmd/taskrun/delete.go

```diff
@@ -60,41 +60,58 @@ or
 				return err
 			}
 
-			if err := opts.CheckOptions(s, args); err != nil {
+			if err := opts.CheckOptions(s, args, p.Namespace()); err != nil {
 				return err
 			}
 
-			return deleteTaskRuns(s, p, args, opts.ParentResourceName)
+			return deleteTaskRuns(s, p, args, opts)
 		},
 	}
 	f.AddFlags(c)
 	c.Flags().BoolVarP(&opts.ForceDelete, "force", "f", false, "Whether to force deletion (default: false)")
 	c.Flags().StringVarP(&opts.ParentResourceName, "task", "t", "", "The name of a task whose taskruns should be deleted (does not delete the task)")
+	c.Flags().BoolVarP(&opts.DeleteAllNs, "all", "", false, "Delete all taskruns in a namespace (default: false)")
 	_ = c.MarkZshCompPositionalArgumentCustom(1, "__tkn_get_taskrun")
 	return c
 }
 
-func deleteTaskRuns(s *cli.Stream, p cli.Params, trNames []string, parentTask string) error {
+func deleteTaskRuns(s *cli.Stream, p cli.Params, trNames []string, opts *options.DeleteOptions) error {
 	cs, err := p.Clients()
 	if err != nil {
 		return fmt.Errorf("failed to create tekton client")
 	}
 	var d *deleter.Deleter
-	if parentTask == "" {
+	switch {
+	case opts.DeleteAllNs:
+		d = deleter.New("TaskRun", func(taskRunName string) error {
+			return cs.Tekton.TektonV1alpha1().TaskRuns(p.Namespace()).Delete(taskRunName, &metav1.DeleteOptions{})
+		})
+		trs, err := allTaskRunNames(p, cs)
+		if err != nil {
+			return err
+		}
+		d.Delete(s, trs)
+	case opts.ParentResourceName == "":
 		d = deleter.New("TaskRun", func(taskRunName string) error {
 			return cs.Tekton.TektonV1alpha1().TaskRuns(p.Namespace()).Delete(taskRunName, &metav1.DeleteOptions{})
 		})
 		d.Delete(s, trNames)
-	} else {
+	default:
 		d = deleter.New("Task", func(_ string) error {
 			return errors.New("the task should not be deleted")
 		})
 		d.WithRelated("TaskRun", taskRunLister(p, cs), func(taskRunName string) error {
 			return cs.Tekton.TektonV1alpha1().TaskRuns(p.Namespace()).Delete(taskRunName, &metav1.DeleteOptions{})
 		})
-		d.DeleteRelated(s, []string{parentTask})
+		d.DeleteRelated(s, []string{opts.ParentResourceName})
+	}
+	if !opts.DeleteAllNs {
+		d.PrintSuccesses(s)
+	} else {
+		if d.Errors() == nil {
+			fmt.Fprintf(s.Out, "All TaskRuns deleted in namespace %q\n", p.Namespace())
```

Comment:
Print message saying all taskruns were deleted instead of showing every taskrun name that was deleted. Only displays if no errors.